tests/spicy/types/result/ops.spicy

```diff
@@ -31,3 +31,9 @@ import spicy;
 global err: spicy::Error;
 err = error"abc";
 assert err.description() == "abc";
+
+global x = True : error"my-error";
+assert x;
```

Comment:
I like how this makes checks with dedicated error messages reusable, and the parser currently even accepts a plain `string` via implicit conversions which I like even better,

```ruby
global reality_check = True: "oh no!";
assert reality_check;
```

This works with string literals, but fails with runtime strings, do you think we could make this work?
```ruby
global oh_my = "oh no!";
global reality_check = True: oh_my; # ERROR: type of error expression must be 'error', but is string
assert reality_check;
```

By not allowing this we break existing code which we should continue to support (it will likely be even better with the new potential for reuse):

```ruby
global x = 42;
assert x % 2 == 0 : ("%d expected to be even" % x);
```

Reply:
> This works with string literals, but fails with runtime strings, do you think we could make this work?

Yes, will do. I actually left it out on purpose but didn't think about `%`.